core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -1040,4 +1047,195 @@ public void testAllManifestsTableSnapshotNot() {
         expectedManifestListPaths(table.snapshots(), 1L, 3L, 4L),
         actualManifestListPaths(manifestsTableScan));
   }
+
+  @Test
+  public void testPositionDeletesWithFilter() {
+    Assume.assumeTrue("Position deletes supported only for v2 tables", formatVersion == 2);
+    preparePartitionedTable();
+
+    PositionDeletesTable positionDeletesTable = new PositionDeletesTable(table);
+
+    Expression expression =
+        Expressions.and(
+            Expressions.equal("partition.data_bucket", 1), Expressions.greaterThan("pos", 0));
+    BatchScan scan = positionDeletesTable.newBatchScan().filter(expression);
+    assertThat(scan).isExactlyInstanceOf(PositionDeletesTable.PositionDeletesBatchScan.class);
+
+    List<ScanTask> tasks = Lists.newArrayList(scan.planFiles());
+
+    Assert.assertEquals(
+        "Expected to scan one delete manifest",
+        1,
+        ((PositionDeletesTable.PositionDeletesBatchScan) scan)
+            .scanMetrics()
+            .scannedDeleteManifests()
+            .value());
+    Assert.assertEquals(
+        "Expected to skip three delete manifests",
+        3,
+        ((PositionDeletesTable.PositionDeletesBatchScan) scan)
+            .scanMetrics()
+            .skippedDeleteManifests()
+            .value());
+
+    assertThat(tasks).hasSize(1);
+
+    ScanTask task = tasks.get(0);
+    assertThat(task).isInstanceOf(PositionDeletesScanTask.class);
+
+    Types.StructType partitionType = Partitioning.partitionType(table);
+    PositionDeletesScanTask posDeleteTask = (PositionDeletesScanTask) task;
+
+    Assert.assertEquals(
+        "Expected correct partition on task",
+        1,
+        (int) posDeleteTask.file().partition().get(0, Integer.class));
+    Assert.assertEquals(
+        "Expected correct partition on constant column",
+        1,
+        (int)
+            ((StructLike)
```

Comment:
Same here and a few other places in this class.

Reply:
Done